arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -944,13 +979,25 @@ Result RocksDBCollection::insert(arangodb::transaction::Methods* trx,
     guard.finish(hasPerformedIntermediateCommit);
   }
 
+  _statistics._rocksdb_insert_msec.count(duration<float,std::micro>(clock::now() - start).count());
   return res;
 }
 
 Result RocksDBCollection::update(transaction::Methods* trx,
                                  velocypack::Slice newSlice,
                                  ManagedDocumentResult& resultMdr, OperationOptions& options,
                                  ManagedDocumentResult& previousMdr) {
+
+  if (options.isSynchronousReplicationFrom.empty()) {
+    ++_statistics._numWrite;
+  } else {
+    ++_statistics._numReplicate;
+  }
+
+  using namespace std::chrono;
+  using clock = steady_clock;
+  auto start = clock::now();
```

Comment:
same here, in case of early exit we only update the counters but not the timers.

Reply:
That is correct was as intended.